.github/workflows/compat-matrix.yaml

```diff
@@ -0,0 +1,39 @@
+name: Matrix compat
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "**.go"
+      - "!test/**" # exclude changes in e2e tests
+      - ".github/workflows/unit-tests.yaml"
+      - "hack/test.sh"
+      - "charts/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+    name: generates automated matrix
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Set up Go 1.21
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+          cache: false
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+      - name: Execute unit tests
+        run: go run hack/compat-matrix/main.go docs/pages/deploying-vclusters/compat-matrix.mdx
+      - name: commit change if needed
+        run: |
+          git diff --quiet && git diff --staged --quiet || git commit -am 'updated compatibility matrix'
+      - name: Push changes
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TO:KEN }}
+        run: git push
```

Comment:
Please create a PR instead of directly pushing to main. 

See this for reference: https://github.com/loft-sh/vcluster/blob/d6187b08543c2b81e2dca794cff6baed3c3f048b/.github/workflows/go-licenses.yaml#L37-L47